src/ephemeris/setup_data_libraries.py

```diff
@@ -150,19 +154,32 @@ def normalize_items(has_items):
     normalize_items(library_def)
 
     if library_def:
-        create_func(gi, library_def)
+        jobs = list(create_func(gi, library_def))
 
-        no_break = True
-        while True:
-            no_break = False
+        job_ids = []
+        if legacy:
             for job in jc.get_jobs():
-                if job['state'] != 'ok':
-                    no_break = True
-            if not no_break:
+                # Fetch all upload job IDs, ignoring complete ones.
+                if job['tool_id'] == 'upload1' and job['state'] not in ('ok', 'error'):
+                    job_ids.append(job['id'])
+
+            # Just have to check that all upload1 jobs are termianl.
+        else:
+            # Otherwise get back an actual list of jobs
+            for job in jobs:
+                if 'jobs' in job:
+                    for subjob in job['jobs']:
+                        job_ids.append(subjob['id'])
+
+        while True:
+            job_states = [jc.get_state(job) in ('ok', 'error') for job in job_ids]
```

Comment:
```suggestion
            job_states = [jc.get_state(job) in ('ok', 'error', 'deleted', '') for job in job_ids]
```
that'd be all the terminal job states Galaxy knows about ... there's also paused which is sort-of terminal, but I guess that can't really happen in upload jobs.